File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        double sum = 0;
        for (int i = 0; i < k; ++i) sum += nums[i];
        double max_sum = sum;
        for (int i = 0; i < nums.size() - k; ++i) {
            sum += nums[k + i] - nums[i];
            max_sum = max(max_sum, sum);
        }
        return max_sum / k;
    }
};

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution(object):
    def findMaxAverage(self, nums, k):
        n = len(nums)
        sum1 = sum(nums[:k])
        max_sum  = sum1
        for i in range(n - k):
            sum1 += nums[k + i] - nums[i]
            max_sum = max(max_sum, sum1)
            print(max_sum)
        return float(max_sum) / k
